clippy_lints/src/lifetimes.rs

```diff
@@ -586,6 +591,24 @@ where
         self.generic_args_depth -= 1;
     }
 
+    fn visit_fn_decl(&mut self, fd: &'tcx FnDecl<'tcx>) -> Self::Result {
+        let mut input_ref_count = 0;
+        for input in fd.inputs {
+            if let TyKind::Ref(lt, mut_ty) = input.kind {
+                if !lt.is_elided() && !lt.is_anonymous() {
+                    break;
+                }
+                input_ref_count += 1;
+                if input_ref_count > 1 || is_self_ty(mut_ty.ty) {
+                    break;
+                }
+            }
+        }
+        self.lifetime_elision_impossible = input_ref_count != 1;
```

Comment:
If this fix is only about lifetimes in returns, should we only restrict this to only cases where the return contains a lifetime?

Reply:
Yes. Done.